face/CustomerCheckinManagement/python/shared_functions.py

```diff
@@ -0,0 +1,296 @@
+    except requests.exceptions.RequestException as e:
+        print(f"Request failed: {e}")
+        return None
+
+# Function to delete a person
+def delete_person(subscription_key, endpoint, person_id, injection_header=None):
+    delete_person_url = f"{endpoint}/face/v1.1-preview.1/persons/{person_id}"
+    headers = {
+        'Ocp-Apim-Subscription-Key': subscription_key,
+        'X-MS-AZSDK-Telemetry': injection_header
+    }
+    try:
+        response = requests.delete(delete_person_url, headers=headers)
+        response.raise_for_status()
+
+        if response.status_code == 202:
+            operation_location = response.headers.get('Operation-Location')
+            if operation_location:
+                if check_operation_status(subscription_key, operation_location):
+                    return True
+                else:
+                    print ("Failed to delete person.")
+                    return False
+            else:
+                print("No Operation-Location header found in the response.")
+                return False
+        else:
+            print(f"Failed to delete person: {response.json()}")
+            return False
+    except requests.exceptions.RequestException as e:
+        print(f"Request failed: {e}")
+        return False
+
+# Function to identify faces in an image
+def identify_faces(subscription_key, endpoint, face_id, dynamic_person_group_id=None, injection_header=None):
+    headers = {
+        'Ocp-Apim-Subscription-Key': subscription_key,
+        'Content-Type': 'application/octet-stream',
+        'X-MS-AZSDK-Telemetry': injection_header
+    }
+
+    identify_url = endpoint + "/face/v1.1-preview.1/identify"
+    headers['Content-Type'] = 'application/json'
+    body = {
+        'faceIds': [face_id],
+        'maxNumOfCandidatesReturned': 1,
+        'confidenceThreshold': 0.5
+    }
+
+    if dynamic_person_group_id:
+        body['dynamicPersonGroupId'] = dynamic_person_group_id
+    else:
+        body['personIds'] = '*'
+
+    response = requests.post(identify_url, headers=headers, json=body)
+    response.raise_for_status()
+    results = response.json()
+
+    if len(results) > 0:
+        return results[0]['candidates'][0]['personId']
+    else:
+        return None
+
+# Function to create a dynamic person group
+def create_dynamic_person_group(subscription_key, endpoint, dynamic_person_group_id, injection_header=None):
+    create_DPG_url = f"{endpoint}/face/v1.1-preview.1/dynamicpersongroups/{dynamic_person_group_id}"
+    headers = {
+        'Ocp-Apim-Subscription-Key': subscription_key,
+        'Content-Type': 'application/json',
+        'X-MS-AZSDK-Telemetry': injection_header
+    }
+    body = {
+        "name": "Example DynamicPersonGroup",
+        "userData": "User defined data",
+    }
+    try:
+        response = requests.put(create_DPG_url, headers=headers, json=body)
+        response.raise_for_status()
+
+        if response.status_code == 202:
```

Comment:
Creating empty DPG is not an async call. Its succeeded status code would be 200 instead of 202, and does not carry the Operation-Location header, either.

ref: https://learn.microsoft.com/en-us/rest/api/face/person-directory-operations/create-dynamic-person-group?view=rest-face-v1.2-preview.1&tabs=HTTP#response